iris-mpc-gpu/src/server/actor.rs

```diff
@@ -340,6 +341,9 @@ impl ServerActor {
         let batch_match_list_left = distance_comparator.prepare_db_match_list(n_queries);
         let batch_match_list_right = distance_comparator.prepare_db_match_list(n_queries);
 
+        let or_policy_bitmap =
+            distance_comparator.prepare_luc_bitmap(max_db_size / device_manager.device_count());
```

Comment:
That's already the more optimized version, we just need to watch out later to also split the ranges for the indices correctly. The most trivial solution would otherwise be to just allocate `max_db_size` on all devices and have the same (unnecessarily) large bitmap on each.